## ðŸ§¬ Introduction to Bioinformatics Final Project

#### Project Overview

This repository is a curated collection of bioinformatics tasks, each offering a distinct perspective on genomic and transcriptomic analysis. Our projects range from the study of neurobiological responses in Zebra Finches, to analyzing the footprint of an acnient microbe in birds and phylogenetic analysis using mitochondrial genomes. These tasks provide an immersive experience in the application of bioinformatics tools and methodologies to answer real-world biological questions.

#### Tasks Overview

1. **Find The Imposter - Deciphering Mysterious Sequences**:
   - Focus: Assembly of a collection of short reads, BLASTing it and finding interesting results.
   - Key Components: De Novo Assembly, Galaxy Project, BLAST.

2. **Comparative Analysis of Mitochondrial Genomes**:
   - Focus: Understand evolutionary relationships between species by analyzing mitochondrial DNA.
   - Key Components: Mitochondrial genome alignment, phylogenetic analysis, and exploration of ecological and anthropological insights.

3. **Neural and Immunological Responses of Zebra Finches**:
   - Focus: Investigate the influence of social environment on neural responses to sickness in Zebra Finches.
   - Key Components: Data preprocessing, differential gene expression analysis, functional annotation, visualization, and interpretation in the context of neurobiology, immunology, and social behavior.

4. **A Single-Cell Perspective on Gene Expression**:
   - Focus: Design a project to show the limitations of RNA-Seq and the effectiveness of single-cell analysis in gene expression studies.
   - Key Components: RNA-Seq data analysis, single-cell sequencing techniques, data interpretation, and creative project design.

#### Hands-On Experience and Freedom in Exploration

The tasks in this repository are designed to provide a hands-on experience in bioinformatics analysis. Participants have full freedom to explore various methodologies, tools, and approaches. The goal is to encourage creativity, foster problem-solving skills, and provide an opportunity to apply theoretical knowledge in practical scenarios.

#### Repository Structure

- Each task is organized into its own directory, complete with necessary datasets and Jupyter notebooks.
- Supplementary materials and references are included, offering guidance and additional learning resources.
- Visualization scripts and example outputs are provided to facilitate and enrich the analysis process.

#### Getting Started

To begin, clone this repository to your local machine and navigate to the individual task directories. Each task is accompanied by a Jupyter notebook that guides you through the analysis process. The notebooks are structured to offer both direction and flexibility, allowing you to delve deep into the data and draw your own conclusions.

#### Resources

This section consists of an extensive list of resources for various tools, techniques, and methods relevant to the tasks in this repository.

#### RNA-Seq Data Analysis

- **DESeq2**:
  - [DESeq2 Official Documentation](https://bioconductor.org/packages/release/bioc/html/DESeq2.html)
  - [DESeq2 Vignette for Detailed Analysis](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html)
  - [DESeq2 GitHub Repository](https://github.com/mikelove/DESeq2)

- **edgeR**:
  - [edgeR User Guide](https://bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf)
  - [Analyzing RNA-seq data with edgeR](https://f1000research.com/articles/5-1438)

#### PCA and t-SNE in Python

- **PCA**:
  - [Scikit-learn PCA Documentation](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html)
  - [PCA in Python Example Notebook](https://github.com/dataprofessor/code/blob/master/python/PCA_analysis.ipynb)

- **t-SNE**:
  - [t-SNE User Guide in Scikit-learn](https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html)
  - [t-SNE Visualization Example Notebook](https://github.com/oreillymedia/t-SNE-tutorial)

#### Phylogenetic Analysis

- **ETE Toolkit**:
  - [ETE Toolkit Tutorial](http://etetoolkit.org/docs/latest/tutorial/tutorial_phylogeny.html)

- **DendroPy**:
  - [DendroPy Documentation](https://dendropy.org/)
  - [Phylogenetic Computing with DendroPy](https://academic.oup.com/sysbio/article/59/3/339/1702852)

- **BEAST**:
  - [BEAST Software](https://www.beast2.org/)
  - [Tutorials for BEAST](https://taming-the-beast.org/tutorials/)

#### Functional Annotation and Pathway Analysis

- **DAVID**:
  - [DAVID Functional Annotation Tool](https://david.ncifcrf.gov/)

- **Gene Ontology (GO)**:
  - [Gene Ontology Consortium](http://geneontology.org/)
  - [GO Analysis Step-by-Step Tutorial](https://www.bioconductor.org/help/course-materials/2011/CSAMA/labs/lab02-GOanalysis/GOanalysis.pdf)

#### Additional Learning Materials

- **Python for Bioinformatics**:
  - [Biopython Tutorial and Cookbook](https://biopython.org/DIST/docs/tutorial/Tutorial.html)
  - [Python for Biologists](https://pythonforbiologists.com/)

- **R for Bioinformatics**:
  - [Bioconductor - Open Source Software for Bioinformatics](https://www.bioconductor.org/)

- **GitHub Repositories and Notebooks**:
  - [Awesome Bioinformatics on GitHub](https://github.com/danielecook/Awesome-Bioinformatics)
